src/js/outreach/views/opt-in_views.js

```diff
@@ -79,8 +113,17 @@ const OptInView = View.extend({
   onChangeLastName() {
     this.model.set({ last_name: trim(this.ui.lastName.val()) });
   },
-  onChangeBirthDate() {
-    this.model.set({ birth_date: trim(this.ui.birthDate.val()) });
+  onChangeDobInput() {
+    const dobMonth = this.ui.dobMonth.val();
+    const dobDay = this.ui.dobDay.val();
+    const dobYear = this.ui.dobYear.val();
+
+    if (!dobMonth || !dobDay || !dobYear) {
+      this.model.set({ birth_date: '' });
+      return;
+    }
+
+    this.model.set({ birth_date: `${ dobYear }-${ dobMonth }-${ dobDay }` });
```

Comment:
We probably need to add leading zeros to any single digit values, right? For it to be valid when sending the dob string to the BE.

Like this: `2025-5-5` => `2025-05-05`

Reply:
yep, but we should probably stick it in dayjs to make sure it's event valid.  Like the 31st of Feb.. the dayjs can format `YYYY-MM-DD`

Reply:
Validating this date while also keeping the submit button in the correct enabled/disabled state was tricky.

Do we need to validate the date? If I remember correctly, this date string is being sent to the BE and compared to what's in the db. It's not saving/inserting any new data into the db, right? 

So an invalid date like `2025-2-31` being sent wouldn't really cause any issues, right?

Reply:
When generating the `dayjs()` date obj and then using `.isValid()` against it, there were a lot of edge cases not covered.

Like both of these are being considered a valid date by `dayjs().isValid()`:

```js
dayjs('-10-10');
dayjs('null-10-10');
```

Since `isValid()` simply checks if a date object could be created.

You can do a strict check like this:

```js
dayjs('1990-02-15', 'YYYY-MM-DD', true).isValid();
```

But then we run into issues because the originally inputted values may or may not have the leading zeros.